pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java

```diff
@@ -86,20 +86,24 @@ public StringRepresentation getTableSize(
     tableSizeInfo.diskSizeInBytes = 0L;
 
     ImmutableList<SegmentDataManager> segmentDataManagers = tableDataManager.acquireAllSegments();
-    for (SegmentDataManager segmentDataManager : segmentDataManagers) {
-      IndexSegment segment = segmentDataManager.getSegment();
-      long segmentSizeBytes = segment.getDiskSizeBytes();
-      if (detailed) {
-        SegmentSizeInfo segmentSizeInfo = new SegmentSizeInfo(segment.getSegmentName(), segmentSizeBytes);
-        tableSizeInfo.segments.add(segmentSizeInfo);
+    try {
+      for (SegmentDataManager segmentDataManager : segmentDataManagers) {
+        IndexSegment segment = segmentDataManager.getSegment();
+        long segmentSizeBytes = segment.getDiskSizeBytes();
+        if (detailed) {
+          SegmentSizeInfo segmentSizeInfo = new SegmentSizeInfo(segment.getSegmentName(), segmentSizeBytes);
+          tableSizeInfo.segments.add(segmentSizeInfo);
+        }
+        tableSizeInfo.diskSizeInBytes += segmentSizeBytes;
```

Comment:
So if we have a segment of 10000 bytes and a defunct segment, `diskSizeInBytes` should be 9999 bytes?